app/api/scorm_extension_comments_api.rb

```diff
@@ -0,0 +1,54 @@
+require 'grape'
+
+class ScormExtensionCommentsApi < Grape::API
+  helpers AuthenticationHelpers
+  helpers AuthorisationHelpers
+
+  desc 'Request a scorm extension for a task'
+  params do
+    requires :comment, type: String, desc: 'The details of the request'
+  end
+  post '/projects/:project_id/task_def_id/:task_definition_id/request_scorm_extension' do
+    project = Project.find(params[:project_id])
+    task_definition = project.unit.task_definitions.find(params[:task_definition_id])
+    task = project.task_for_task_definition(task_definition)
+
+    # check permissions using specific permission has with addition of request extension if allowed in unit
+    unless authorise? current_user, task, :request_scorm_extension
+      error!({ error: 'Not authorised to request a scorm extension for this task' }, 403)
+    end
+
+    if task_definition.scorm_attempt_limit == 0
+      error!({ message: 'This task allows unlimited attempts to complete the test' }, 400)
+      return
```

Comment:
error! throws an exception - so this is unreachable... we should look and remove these unnecessary return statements